paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerSessionSavedSelectionDataSource.kt

```diff
@@ -1,14 +1,49 @@
 package com.stripe.android.customersheet.data
 
+import com.stripe.android.core.injection.IOContext
+import com.stripe.android.paymentsheet.PrefsRepository
 import com.stripe.android.paymentsheet.model.SavedSelection
+import kotlinx.coroutines.withContext
+import java.io.IOException
 import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
 
-internal class CustomerSessionSavedSelectionDataSource @Inject constructor() : CustomerSheetSavedSelectionDataSource {
+internal class CustomerSessionSavedSelectionDataSource @Inject constructor(
+    private val elementsSessionManager: CustomerSessionElementsSessionManager,
+    private val prefsRepositoryFactory: @JvmSuppressWildcards (String) -> PrefsRepository,
+    @IOContext private val workContext: CoroutineContext,
+) : CustomerSheetSavedSelectionDataSource {
     override suspend fun retrieveSavedSelection(): CustomerSheetDataResult<SavedSelection?> {
-        throw NotImplementedError("Not implemented yet!")
+        return withContext(workContext) {
+            createPrefsRepository().mapCatching { prefsRepository ->
+                prefsRepository.getSavedSelection(
+                    /*
+                     * We don't calculate on `Google Pay` availability in this function. Instead, we check
+                     * within `CustomerSheet` similar to how we check if a saved payment option is still exists
+                     * within the user's payment methods from `retrievePaymentMethods`
+                     */
+                    isGooglePayAvailable = true,
+                    isLinkAvailable = false,
```

Comment:
Why is `isLinkAvailable` false here? Do we not support Link in Customer Sheet?

Reply:
We do not at the moment. I don't think there are plans to support `Link` for `CustomerSheet`.